src/index.js

```diff
@@ -1,38 +1,97 @@
-      this.getMenuElementsAndAddEventsByDataAttrs()
+      this.getMenuElementsAndAddEvents();
+      this.getMenuElementsAndAddEventsByDataAttrs();
+      this.getNavbarElementsAndAddEventsByDataThqAttrs();
       return this;
     };
 
     getMenuElementsAndAddEventsByDataAttrs = () => {
-      const allHeaders = teleport.getAllElementsByDataAttribute('role', 'Header')
+      const allHeaders = teleport.getAllElementsByDataAttribute(
+        "role",
+        "Header"
+      );
 
-      if (!allHeaders.length) {
-        teleport.log("No teleport Headers found in your project")
+      if (allHeaders.length === 0) {
+        teleport.log("No teleport Headers found in your project");
       }
 
-      allHeaders.forEach(header => {
-        const burgerBtn = teleport.getElByDataAttribute('type', 'BurgerMenu', header)
-        const mobileMenu = teleport.getElByDataAttribute('type', 'MobileMenu', header)
-        const closeBtn = teleport.getElByDataAttribute('type', 'CloseMobileMenu', header)
+      allHeaders.forEach((header) => {
+        const burgerBtn = teleport.getElByDataAttribute(
+          "type",
+          "BurgerMenu",
+          header
+        );
+        const mobileMenu = teleport.getElByDataAttribute(
+          "type",
+          "MobileMenu",
+          header
+        );
+        const closeBtn = teleport.getElByDataAttribute(
+          "type",
+          "CloseMobileMenu",
+          header
+        );
+
+        burgerBtn.addEventListener("click", () => {
+          mobileMenu.classList.add("teleport-show");
+        });
+
+        closeBtn.addEventListener("click", () => {
+          mobileMenu.classList.remove("teleport-show");
+        });
+      });
+    };
+
+    getNavbarElementsAndAddEventsByDataThqAttrs = () => {
+      const allNavbars = teleport.getAllElementsByDataAttribute(
+        "thq",
+        "thq-navbar"
+      );
 
-        burgerBtn.addEventListener('click', () => {
-          mobileMenu.classList.add('teleport-show')
-        })
+      if (allNavbars.length === 0) {
+        teleport.log("No teleport Navbars found in your project");
+      }
+
+      allNavbars.forEach((navbar) => {
+        const burgerBtn = teleport.getElByDataAttribute(
+          "thq",
+          "thq-burger-menu",
+          navbar
+        );
+        const mobileMenu = teleport.getElByDataAttribute(
+          "thq",
+          "thq-mobile-menu",
+          navbar
+        );
+        const closeBtn = teleport.getElByDataAttribute(
+          "thq",
+          "thq-close-menu",
+          navbar
+        );
 
-        closeBtn.addEventListener('click', () => {
-          mobileMenu.classList.remove('teleport-show')
-        })
-      })
+        if (!burgerBtn || !mobileMenu || !closeBtn) {
+          teleport.error(
+            "The navbar elements (burger button, mobile menu, close button) could not be found.",
+            navbar
+          );
+        }
+
+        burgerBtn.addEventListener("click", () => {
```

Comment:
I assume those will crash  if the burgerBtn or closeBtn are undefined, maybe a return in the if statment above will fix, pls check it